src/evaluators/utility/PHAL_SharedParameter.hpp

```diff
@@ -32,8 +32,61 @@ class SharedParameter : public PHX::EvaluatorWithBaseImpl<Traits>,
 
     // Sacado-ized parameter
     Teuchos::RCP<ParamLib> paramLib = p.get<Teuchos::RCP<ParamLib>>("Parameter Library");
-    this->registerSacadoParameter(param_name, paramLib);
     this->setName("Shared Parameter " + param_name + PHX::print<EvalT>());
+
+    const Teuchos::ParameterList* paramsList = p.get<const Teuchos::ParameterList*>("Parameters List");
+
+    // Find the parameter in the Paramter list,
+    // register as a Sacado Parameter and set the Nominal value
+    bool nominalValueSet = false;
+    if((paramsList != NULL) && paramsList->isParameter("Number Of Parameters"))
```

Comment:
Very minor: in C++ is more apt to use `nullptr` instead of the C-style `NULL`.

Reply:
OK, I'll fix that.